src/Components/Peoples/Peoples.tsx

```diff
@@ -0,0 +1,180 @@
+    debounce((newQuery: string) => {
+      if (newQuery) {
+        searchParams.set('query', newQuery);
+      } else {
+        searchParams.delete('query');
+      }
+
+      navigate({
+        search: searchParams.toString(),
+      });
+    }, 500),
+    [],
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+
+    setQuery(value);
+    applyQuery(value);
+  };
+
+  const sortOrder = searchParams.get('sortOrder')?.toString();
+  const sortBy = searchParams.get('sortBy')?.toString();
+
+  const handleSortChage = (value: string) => {
+    if (sortOrder === 'asc' && sortBy === value) {
+      searchParams.set('sortOrder', 'desc');
+    } else {
+      searchParams.set('sortOrder', 'asc');
+    }
+
+    searchParams.set('sortBy', value);
+
+    navigate({
+      search: searchParams.toString(),
+    });
+  };
+
+  const sortClassNames = (value: string) => {
+    if (sortOrder === 'desc' && sortBy === value) {
+      return 'fa-sort-up';
+    }
+
+    return 'fa-sort-down';
+  };
+
+  if (sortBy) {
+    visiblePeople = visiblePeople.sort((a, b): number => {
+      switch (sortBy) {
+        case 'name':
+        case 'sex':
+          return sortOrder === 'asc'
+            ? a[sortBy].localeCompare(b[sortBy])
+            : b[sortBy].localeCompare(a[sortBy]);
+        case 'born':
+        case 'died':
+          return sortOrder === 'asc'
+            ? +a[sortBy] - +b[sortBy]
+            : +b[sortBy] - +a[sortBy];
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return (
+    <div className="peoples">
+      <h3 className="title is-3">People</h3>
+
+      <div className="table-header">
+        <input
+          className="input is-normal"
+          type="search"
+          value={query}
+          placeholder="Filter people"
+          onChange={handleQueryChange}
+        />
+
+        <NavLink
+          to="/people/new"
+          type="button"
+          className="button is-normal"
+        >
+          Add a person
+        </NavLink>
+      </div>
+
+      <Routes>
+        <Route path="/new" element={<NewPerson people={people} setPeople={setPeople} />} />
+      </Routes>
+
+      <table className="table is-narrow">
+        <thead>
+          <tr>
+            <th
+              className="table__header-link"
+              onClick={() => {
+                handleSortChage('name');
+              }}
+            >
+              Name
+              <i className={`fas ${sortClassNames('name')}`} />
+            </th>
+
+            <th
+              className="table__header-link"
+              onClick={() => {
+                handleSortChage('sex');
+              }}
+            >
+              Sex
+              <i className={`fas ${sortClassNames('sex')}`} />
+            </th>
+
+            <th
+              className="table__header-link"
+              onClick={() => {
+                handleSortChage('born');
```

Comment:
```suggestion
                handleSortChage('born');
```
Можна було б створити enum для name, sex, born, died